airflow/providers/amazon/aws/hooks/batch_client.py

```diff
@@ -0,0 +1,503 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""
+A client for AWS batch services
+
+.. seealso::
+
+    - http://boto3.readthedocs.io/en/latest/guide/configuration.html
+    - http://boto3.readthedocs.io/en/latest/reference/services/batch.html
+    - https://docs.aws.amazon.com/batch/latest/APIReference/Welcome.html
+"""
+
+from random import uniform
+from time import sleep
+from typing import Optional
+
+import botocore.client
+import botocore.exceptions
+from typing_extensions import Protocol
+
+from airflow import AirflowException, LoggingMixin
+from airflow.contrib.hooks.aws_hook import AwsHook
+
+# pylint: disable=invalid-name, unused-argument
+
+
+class BatchProtocol(Protocol):
+    """
+    A typing extension for ``boto3.client('batch') -> botocore.client.Batch``
+    """
+
+    def describe_jobs(self, jobs):
+        """Get job descriptions from AWS batch
+
+        :param jobs: a list of JobId to describe
+        :type jobs: List[str]
+
+        :return: an API response to describe jobs
+        :rtype: dict
+        """
+        ...
+
+    def get_waiter(self, waiter_name):
```

Comment:
```suggestion
    def get_waiter(self, waiterName):
```
Please be consistent.

Reply:
I think we should stick to snake case to be more pythonic.

Reply:
Obviously, there's an inconsistency with `boto3` and `botocore` and, in various ways, those libraries are not pythonic, probably due to overuse of JSON and dynamic metaprogramming or something.  In Airflow, even if a method is a wrapper or typing extension, I would prefer to use snake case.  But, if a typing extension (protocol) must exactly match the `boto3` or `botocore` signature, I'm OK with changing this and adding the required pylint comments (if not already there).

Reply:
Given that `submit_job` is already matching the `boto3` or `botocore` parameter names, I've decided to follow that pattern for `get_waiter(waiterName)` also.